codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -233,13 +241,34 @@ private static FlowState state(Http2Stream stream) {
         private int lowerBound;
         private boolean endOfStream;
 
-        FlowState(Http2Stream stream, int initialWindowSize) {
+        /**
+         * The associated {@link #stream} has transitioned into a state which requires the full amount of
+         * state for flow control.
+         */
+        void streamActivated() {
+            // TODO (scottmitch) should this object act as a proxy object to another object that holds the state?
```

Comment:
The idea here is streams that are in states where it is not possible to send/receive flow controlled frames may not need the full set of state that a stream which can have flow controlled frames exchanged.  The `null it out` comment is in reference to if we can extract that state into an object, and only have it allocated/referenced when we actually need it then it can be reclaimed by GC.